File: elixir/7ku/the_poet_and_the_pendulum.ex
defmodule Pendulum do
  def pendulum(values) do
    sorted =
      values
      |> Enum.sort()
      |> Enum.with_index()

    arr1 =
      sorted
      |> Enum.map(fn {x, y} ->
        if rem(y, 2) == 0, do: x
      end)
      |> Enum.reject(&is_nil(&1))

    arr2 =
      sorted
      |> Enum.map(fn {x, y} ->
        if rem(y, 2) != 0, do: x
      end)
      |> Enum.reject(&is_nil(&1))

    arr1
    |> Enum.reverse()
    |> Enum.concat(arr2)
  end
end

File: ruby/7ku/The Poet And The Pendulum.rb
# frozen_string_literal: true
# The Poet And The Pendulum
# https://www.codewars.com/kata/5bd776533a7e2720c40000e5/train/ruby

def pendulum values
  left, right = values.sort.partition.with_index { |_, i| i.even? }
  left.reverse + right
end

p pendulum([9, 4, 6, 4, 10, 5]) # [9, 5, 4, 4, 6, 10]